Add support for environment variables for creds & cross-OS support

dylanb124 opened the pull request:
**Cross-OS Support for the Discord Application**
Instead of checking for Discord.exe (which is typically the name of the application running on Windows) it will check for the existence of Discord RPC. This allows the script to work with Mac and Linux, which have different names for the Discord app.

**Environment variable support for credentials**
Some users may prefer to store their credentials as environment variables. This adds support for that, skipping this step of the config setup if environment variables are set. The values do not get written to config if set as an environment variable. A note has been added to the README.

**Discord Client Error-handling**
I noticed that if the Discord client quits or is RPC is otherwise not available, the script crashes and quits. I added error-handling logic to gracefully enter the reconnect loop for RPC if it becomes unavailable, allowing it to continue to run in the background.

Side note -- thanks for your work on this project, hoping you don't mind some small contributions!

dylanb124 committed:
Add support for environment variables for creds & cross-OS support

Clipph reviewed (commented):
Hello! I was just busy so much these days hahahaha I'll test it once my sched gets less hectic.   Thank you for the modifications! I checked the changes and they all looked okay.

Clipph committed:
Update RARPC.py

I noticed that there is also a need to check again for rpc availability in this part because sometimes, when the discord app got closed right after it check its connectivity and then it updates presence, it will be stuck in a loop of trying to update it without checking again if the pipe can be connected or not. I appreciate your concerns about error catching.
